conans/client/command.py

```diff
@@ -199,6 +199,7 @@ def new(self, *args):
                             help='Generate a .gitignore with the known patterns to excluded')
         parser.add_argument("-ciu", "--ci-upload-url",
                             help='Define URL of the repository to upload')
+        parser.add_argument('-d', '--define', action='append')
```

Comment:
Sure this doesn't need the ``Extender`` custom action?

Reply:
sorry, I am not sure I understand, what do you mean... could you possibly describe your question in more details, e.g. with some examples?

Reply:
We have Extender class in our command.py file: https://github.com/conan-io/conan/blob/beaca4195d700c944f0f7b098e267a509f78acb4/conans/client/command.py#L34

It is basically there to implement the ``append`` action it seems. Is the Extender class legacy code not needed? If that is the case maybe we can get rid of it?

Reply:
seems so, at least `append` exists in 2.7, so `Extender` seems to be a redundant bicycle for me, could be removed in conan 2.0

Reply:
If it is really redundant, then let's drop it today, for next 1.35? Of course if we are relatively sure this will not be breaking users, but if we are, no need to wait until Conan 2.0.

This looks good, then, merging.